pagerduty/config.go

```diff
@@ -40,8 +46,16 @@ Please see https://www.terraform.io/docs/providers/pagerduty/index.html
 for more information on providing credentials for this provider.
 `
 
-// Client returns a new PagerDuty client
+// Client returns a PagerDuty client, initializing when necessary.
 func (c *Config) Client() (*pagerduty.Client, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```

Comment:
haven't used mutexes before in Go, but pleasantly surprised how simple this looks! that's all it takes to handle "a bunch of different objects ask for a client at the same time, and we only want to make one, and have all my callers wait until i make that and give the single one to everyone" ? :o cooool